File: src/main/java/com/leijendary/spring/iamtemplate/generator/RandomGenerator.java
package com.leijendary.spring.iamtemplate.generator;

import java.util.Random;

import static java.lang.Character.getNumericValue;

public class RandomGenerator {

    public static final Random RANDOM = new Random();
    public static final String NUMBERS = "0123456789";

    public static String otp() {
    }

    public static int[] digits(int length) {
        final var values = new int[length];

        for (int i = 0; i < length; i++) {
            final var index = RANDOM.nextInt(length);
            final var current = NUMBERS.charAt(index);

            values[i] = getNumericValue(current);
        }

        return values;
    }
}

File: src/main/kotlin/com/leijendary/spring/template/iam/generator/RandomGenerator.kt
package com.leijendary.spring.template.iam.generator

import java.util.*

object RandomGenerator {
    private val RANDOM = Random()
    private const val NUMBERS = "0123456789"

    fun digits(length: Int): IntArray {
        val values = IntArray(length)

        for (i in 0 until length) {
            val index = RANDOM.nextInt(length)
            val current = NUMBERS[index]

            values[i] = Character.getNumericValue(current)
        }

        return values
    }
}
